### PR TITLE
[v1.1][ISSUE-234] Add jupyter service in pytorch1.12 docker

### PR DESCRIPTION
**Pull request by zigzagcai**
### What changes were proposed in this pull request?

add jupyter support in pytorch1.12 docker
### Why are the changes needed?

add jupyter support in pytorch1.12 docker

### How was this patch tested?

UT

### CODE REVIEW

**Review by Jian-Zhang — DISMISSED** [review]
This PR only  includes the notebook in the docker, what about the doc part?

**Comment by Jian-Zhang** [comment]
After combine pytorch110 & pytorch112, do we still need those docker files? 

How about the Readme file in docker folder? Please reflect the changes/steps to start notebook in readme as well.

**Comment by zigzagcai** [comment]
> This PR only includes the notebook in the docker, what about the doc part?

The doc part will be added later.

**Comment by zigzagcai** [comment]
> After combine pytorch110 & pytorch112, do we still need those docker files?
> 
> How about the Readme file in docker folder? Please reflect the changes/steps to start notebook in readme as well.

This PR mainly about fixing the corresponding README when combing PyTorch110 and PyTorch112

File: Dockerfile-ubuntu18.04/DockerfilePytorch112 [hunk 1]
```diff
@@ -39,3 +40,7 @@ RUN conda init bash
 RUN echo "source /opt/intel/oneapi/setvars.sh --force" >> /root/.bashrc
 RUN echo "root:docker" | chpasswd
 ENTRYPOINT [""]
+
+ADD ./notebook/start-notebook.sh /root
+RUN chmod a+x /root/start-notebook.sh
+CMD ["/root/start-notebook.sh"]
```
[the remark this answers is not in the document]
Comment (zigzagcai) [reply Dockerfile-ubuntu18.04/DockerfilePytorch112:48]: I have refined the README.md in Dockerfile directory.
If user want to run docker container with notebook only, they can run `docker run -it --privileged --network host --device=/dev/dri e2eaiok-pytorch112`

**Comment by zigzagcai** [comment]
Completed and ready for merge.,

**Review by Jian-Zhang — COMMENTED** [review]
LGTM.

File: Dockerfile-ubuntu18.04/README.md [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+``` bash
+git clone https://github.com/intel/e2eAIOK.git
+git submodule update --init --recursive
+cd e2eAIOK; python scripts/start_e2eaiok_docker.py --backend [tensorflow, pytorch, pytorch112] --dataset_path ../ --workers host1, host2, host3, host4 --proxy "http://addr:ip"
+```
+
+### To start a notebook directly
+```
+docker run -it --privileged --network host --device=/dev/dri e2eaiok-pytorch112
+```
```
Comment (xuechendi) [inline Dockerfile-ubuntu18.04/README.md:33]: can you also embed this into start_e2eaiok_docker.py? I found by instructing user with cmdline, lots of arguments are missing
see below example, we need to set shm-size, set dataset_path, set workplace set docker_name

["docker", "run",  "--shm-size=100g", "--privileged",  "--network",  "host", "--device=/dev/dri", "-d", "-v", f"{dataset_path}/:/home/vmagent/app/dataset", "-v", f"{current_folder}/:/home/vmagent/app/e2eaiok",  "-w",  "/home/vmagent/app/", "--name", docker_nickname,  docker_name, "/bin/bash", "-c", "\"service ssh start & sleep infinity\""]

File: Dockerfile-ubuntu18.04/DockerfilePytorch112 (hunk 1 above)
[the remark this answers is not in the document]
Comment (xuechendi) [reply Dockerfile-ubuntu18.04/DockerfilePytorch112:48]: @zigzagcai , I think instruct user to use `docker run` is a fallback. You left over a lot of arguments setting. So user won't be able to define dataset_path, and shm_size is also not set here.

[the remark this answers is not in the document]
Comment (xuechendi) [reply Dockerfile-ubuntu18.04/DockerfilePytorch112:48]: @zigzagcai , I tested this PR locally, it works great, please fix README and using start_e2eaiok_docker.py to trigger jupyter and I'll merge it.
I think the cmdline can be something like `python scripts/start_e2eaiok_docker.py --backend [tensorflow, pytorch, pytorch112] --jupyter_mode --dataset_path ../ --workers host1, host2, host3, host4 --proxy "http://addr:ip"`

File: Dockerfile-ubuntu18.04/README.md [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
 $ docker build -t e2eaiok-pytorch112 . -f DockerfilePytorch112 --build-arg http_proxy=http://proxy-ip:proxy-port --build-arg https_proxy=http://proxy-ip:proxy-port
 ```
+
+## How to use the image.
+
+### To start a cluster via one-click script
+``` bash
+git clone https://github.com/intel/e2eAIOK.git
+git submodule update --init --recursive
+cd e2eAIOK; python scripts/start_e2eaiok_docker.py --backend [tensorflow, pytorch, pytorch112] --dataset_path ../ --workers host1, host2, host3, host4 --proxy "http://addr:ip"
```
[the remark this answers is not in the document]
Comment (xuechendi) [reply Dockerfile-ubuntu18.04/README.md:27]: I found there is same issue in main readme, can you fix that as well, thanks

File: Dockerfile-ubuntu18.04/README.md (hunk 2 above)
Comment (xuechendi) [reply Dockerfile-ubuntu18.04/README.md:33]: I think the cmdline can be something like `python scripts/start_e2eaiok_docker.py --backend [tensorflow, pytorch, pytorch112] --jupyter_mode --dataset_path ../ --workers host1, host2, host3, host4 --proxy "http://addr:ip"`

**Comment by xuechendi** [comment]
the script is tested locally, working well. Some small issues still need fixing
modification requested:
1. fix README 
2. using start_e2eaiok_docker.py to trigger jupyter

File: Dockerfile-ubuntu18.04/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (zigzagcai) [reply Dockerfile-ubuntu18.04/README.md]: Got it. It has been fixed the both README.md

File: Dockerfile-ubuntu18.04/README.md (hunk 2 above)
Comment (zigzagcai) [reply Dockerfile-ubuntu18.04/README.md]: Refined.
Can be triggered through this option `jupyter_mode`

python scripts/start_e2eaiok_docker.py --backend [tensorflow, pytorch, pytorch112] --jupyter_mode --dataset_path ../ --workers host1, host2, host3, host4 --proxy "http://addr:ip"

**Comment by zigzagcai** [comment]
> the script is tested locally, working well. Some small issues still need fixing modification requested:
> 
> 1. fix README
> 2. using start_e2eaiok_docker.py to trigger jupyter

All solved.

**Review by xuechendi — APPROVED** [review]
LGTM

**Pull request merged**